prov/gni/src/gnix_cq.c

```diff
@@ -162,11 +162,11 @@ static inline void cq_enqueue(struct gnix_fid_cq *cq,
 
 	fastlock_acquire(&cq->lock);
 
-	/*
-	 * TODO: Handle wait signal support.
-	 */
 	slist_insert_tail(&event->item, &cq->ev_queue);
 
+	if (cq->wait)
+		_gnix_signal_wait_obj(cq->wait);
+
 	fastlock_release(&cq->lock);
```

Comment:
would it be better to release the lock after inserting into the tail and then doing the signal?